ai_traineree/agents/dummy.py

```diff
@@ -0,0 +1,178 @@
+import copy
+import numpy as np
+from ai_traineree.agents import AgentBase
+from ai_traineree.types import AgentState, BufferState, DataSpace, NetworkState
+from typing import Callable, Dict, Optional, Type
+from ai_traineree.loggers import DataLogger
+from ai_traineree.experience import Experience
+
+import warnings
+
+class DummyAgent(AgentBase):
+    """Deep Q-Learning Network (DQN).
+     
+     Agent that returns random values regardless of the input
+     Print mean of the taken actions, observables and reward
+     
+     """
+    model = "Dummy"
+    def __init__(        self,
+        obs_space: DataSpace,
+        action_space: DataSpace,
+        state_transform: Optional[Callable] = None,
+        reward_transform: Optional[Callable] = None,
+        **kwargs
+    ):
+        """Initiates Dummy agent.
+
+        Parameters:
+            obs_space (DataSpace): Dataspace describing the input.
+            action_space (DataSpace): Dataspace describing the output.
+            state_transform (optional func): Function to transform (encode) state before used by the network.
+            reward_transform (optional func): Function to transform reward before use.
+
+        """
+        super().__init__(**kwargs)
+        self.obs_space = obs_space
+        self.action_space = action_space
+        self.iteration: int = 0
+
+        self._loss: float = float("nan")
+
+    @property
+    def loss(self):
+        return {"loss": self._loss}
+
+
+    def reset(self):
+        warnings.warn("Dummy Agent is for debugging purposes only. Nothing to reset.")
+        return
+
+
+    def step(self, exp: None):
+        """Letting the agent to take a step. In this case, since the actions are totally random we don't need to save any observation
+
+        Parameters:
+            obs (ObservationType): Observation.
+            action (int): Discrete action associated with observation.
+            reward (float): Reward obtained for taking action at state.
+            next_obs (ObservationType): Observation in a state where the action took.
+            done: (bool) Whether in terminal (end of episode) state.
```

Comment:
Guess that this is copied from a different agent. My bad, should have updated it one moving to Experience! :P Thank you for the reminder!

Reply:
I changed to those parameters that I used in the agent. Please let me know if you want me to change it ;)